frontend/src/modules/Shares/components/ShareBoxList.js

```diff
@@ -0,0 +1,489 @@
+
+  const fetchOutboxItems = useCallback(async () => {
+    await client
+      .query(
+        getShareRequestsFromMe({
+          filter: {
+            ...filter
+          }
+        })
+      )
+      .then((response) => {
+        setItems(response.data.getShareRequestsFromMe);
+      })
+      .catch((error) => {
+        dispatch({ type: SET_ERROR, error: error.Error });
+      })
+      .finally(() => setLoading(false));
+  }, [client, dispatch, filter]);
+
+  const fetchInboxItems = useCallback(async () => {
+    await client
+      .query(
+        getShareRequestsToMe({
+          filter: {
+            ...filter
+          }
+        })
+      )
+      .then((response) => {
+        setItems(response.data.getShareRequestsToMe);
+      })
+      .catch((error) => {
+        dispatch({ type: SET_ERROR, error: error.Error });
+      })
+      .finally(() => setLoading(false));
+  }, [client, dispatch, filter]);
+
+  function fetchItems() {
+    if (tab === 'inbox') {
+      if (dataset) {
+        fetchDatasetItems().catch((error) => {
+          dispatch({ type: SET_ERROR, error: error.message });
+        });
+      } else {
+        fetchInboxItems().catch((error) => {
+          dispatch({ type: SET_ERROR, error: error.message });
+        });
+      }
+    } else {
+      fetchOutboxItems().catch((error) => {
+        dispatch({ type: SET_ERROR, error: error.message });
+      });
+    }
+  }
+
+  const fetchMyGroupsOptions = useCallback(async () => {
+    await client
+      .query(
+        listAllGroups({
+          filter: Defaults.selectListFilter
+        })
+      )
+      .then((response) => {
+        if (tab === 'inbox') {
+          setDatasetGroupOptions(
+            response.data.listAllGroups.nodes.map((node) => node.groupUri)
+          );
+        } else {
+          setRequestGroupOptions(
+            response.data.listAllGroups.nodes.map((node) => node.groupUri)
+          );
+        }
+      })
+      .catch((error) => {
+        dispatch({ type: SET_ERROR, error: error.Error });
+      })
+      .finally(() => setLoading(false));
+  }, [client, dispatch, tab]);
+
+  const fetchInboxGroupOptions = useCallback(async () => {
+    await client
+      .query(
+        getShareRequestsToMe({
+          filter: Defaults.selectListFilter
+        })
+      )
+      .then((response) => {
+        setRequestGroupOptions(
+          Array.from(
+            new Set(
+              response.data.getShareRequestsToMe.nodes.map(
+                (node) => node.principal.SamlGroupName
+              )
+            )
+          )
+        );
+      })
+      .catch((error) => {
+        dispatch({ type: SET_ERROR, error: error.Error });
+      })
+      .finally(() => setLoading(false));
+  }, [client, dispatch]);
+
+  const fetchInboxDatasetsOptions = useCallback(async () => {
+    setLoading(true);
+    const response = await client.query(
+      listDatasets(Defaults.selectListFilter)
```

Comment:
To test (bc I may be missing something) - but when we call `listDatasets` would we be returning all datasets that the user is Admin or steward to + has requested via other ShareRequests (this second piece we would not want)

Reply:
Resolved in the latest commits